.yarnrc.yml

```diff
@@ -17,13 +17,23 @@ yarnPath: .yarn/releases/yarn-3.3.0.cjs
 
 packageExtensions:
   "@craco/craco@*":
+    dependencies:
+      # postcss is included already in the `react-scripts` package
+      postcss: "^8.4.4"
```

Comment:
@a-ignatov-parc I have taken the version from `react-scripts` package. Should I replace the verion with `"*"` ?
`postcss: "*"`

Reply:
@r-bartolomei, it's always better to rely on the `peerDependencies` constraints, otherwise we can install a version that is older or newer than supported.

Here is how the constraints can be determined.

Let's remove `packageExtensions` and see what Yarn will tell us.

```shell
» yarn
➤ YN0000: ┌ Resolution step
➤ YN0002: │ @craco/craco@npm:7.1.0 [28403] doesn't provide @types/node (p60610), requested by cosmiconfig-typescript-loader
➤ YN0002: │ @craco/craco@npm:7.1.0 [28403] doesn't provide postcss (p2044b), requested by autoprefixer
➤ YN0002: │ @craco/craco@npm:7.1.0 [28403] doesn't provide typescript (p6ec49), requested by cosmiconfig-typescript-loader
➤ YN0002: │ @craco/types@npm:7.1.0 doesn't provide eslint (pc375d), requested by eslint-webpack-plugin
➤ YN0002: │ @craco/types@npm:7.1.0 doesn't provide postcss (pc4136), requested by autoprefixer
➤ YN0002: │ eslint-config-react-app@npm:7.0.1 [b5c1c] doesn't provide @babel/plugin-syntax-flow (pa345c), requested by eslint-plugin-flowtype
➤ YN0002: │ eslint-config-react-app@npm:7.0.1 [b5c1c] doesn't provide @babel/plugin-transform-react-jsx (p54bce), requested by eslint-plugin-flowtype
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0000: └ Completed in 0s 233ms
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed in 0s 414ms
➤ YN0000: Failed with errors in 0s 944ms
------------------------------------------------------------
```

The lines we're interested in are the following ones:
```shell
➤ YN0002: │ @craco/craco@npm:7.1.0 [28403] doesn't provide postcss (p2044b), requested by autoprefixer
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
```

Here's what we get if we run the following command

```shell
» yarn explain peer-requirements p2044b
➤ YN0000: @craco/craco@npm:7.1.0 [28403] doesn't provide postcss, breaking the following requirements:

➤ YN0000: autoprefixer@npm:10.4.14 [693f5] → ^8.1.0 ✘
------------------------------------------------------------
```

And `postcss: "^8.1.0"` will be the correct value for `packageExtensions`.

@r-bartolomei, please check all constraints as I explained above and update the `.yarnrc.yml` accordingly 👍

Reply:
> it's always better to rely on the `peerDependencies` constraints

@a-ignatov-parc if I add `peerDependencies` this means the package will be checked if it exist on the parent package, in our case the `@truvity/webapp-playground`:
```
yarn
➤ YN0000: ┌ Resolution step
➤ YN0002: │ @truvity/webapp-playground@workspace:. doesn't provide eslint (pdb773), requested by @craco/craco
➤ YN0002: │ @truvity/webapp-playground@workspace:. doesn't provide eslint (pac5b2), requested by @craco/types
➤ YN0002: │ @truvity/webapp-playground@workspace:. doesn't provide postcss (pe3ac0), requested by @craco/craco
➤ YN0002: │ @truvity/webapp-playground@workspace:. doesn't provide postcss (pbc0a3), requested by @craco/types
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed
```

If I keep `dependencies` it'll check on other packages as well as i understand.

I have made all the changes you've requested based on versions mentioned in `yarn explain peer-requirements`

Reply:
@r-bartolomei, that's expected. We should not extend the package with `peerDependencies` that we can't provide in our root package. 

May I ask why you updated `postcss` from `dependencies` to `peerDependencies`? If this is due to the following phrase, then you have misunderstood its goal.

> it's always better to rely on the `peerDependencies` constraints

With that phrase, I tried to explain from where to get version values for extension, not how to define them.

Reply:
@r-bartolomei, if things have not cleared up let's have a call